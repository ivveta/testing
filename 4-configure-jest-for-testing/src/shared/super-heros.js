const superHeros = [
  {name: 'Dynagay', powers: ['disintegration ray', 'fly']},
  {name: 'Apogee', powers: ['gravity control', 'fly']},
  {name: 'Blazestone', powers: ['control of fire', 'pyrotechnic discharges']},
  {name: 'Frozone', powers: ['freeze water']},
  {name: 'Jack-Jack', powers: ['shapeshifting', 'fly']},
]

function getFlyingSuperHeros() {
  return superHeros.filter(hero => {
    return hero.powers.includes('fly')
  })
}

export {getFlyingSuperHeros}