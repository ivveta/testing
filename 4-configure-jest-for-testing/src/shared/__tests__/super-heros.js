import {getFlyingSuperHeros} from '../super-heros'

test('returns super heros that can fly', () => {
  const flyingHeros = getFlyingSuperHeros()

  expect(flyingHeros).toMatchInlineSnapshot(`
[
  {
    "name": "Dynagay",
    "powers": [
      "disintegration ray",
      "fly",
    ],
  },
  {
    "name": "Apogee",
    "powers": [
      "gravity control",
      "fly",
    ],
  },
  {
    "name": "Jack-Jack",
    "powers": [
      "shapeshifting",
      "fly",
    ],
  },
]
`)
})