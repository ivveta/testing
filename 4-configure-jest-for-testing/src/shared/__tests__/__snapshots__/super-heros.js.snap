// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns super heros that can fly 1`] = `
[
  {
    "name": "Dynagay",
    "powers": [
      "disintegration ray",
      "fly",
    ],
  },
  {
    "name": "Apogee",
    "powers": [
      "gravity control",
      "fly",
    ],
  },
  {
    "name": "Jack-Jack",
    "powers": [
      "shapeshifting",
      "fly",
    ],
  },
]
`;
